src/input.rs

```diff
@@ -42,6 +42,7 @@ struct IsConstant(bool);
 impl<DB, Q> InputStorage<DB, Q>
 where
     Q: Query<DB>,
+    Q::Value: Eq,
```

Comment:
Would this `untracked` policy replace one of the following enums: `AlwaysMemoizeValue`, `NeverMemoizeValue`, `VolatileValue`? They are all unique in with respect to memoizing and tracking inputs, so I did not follow what you mean by that.

I needed to move the `Eq` bounds into the input queries as well. You mentioned:

> we might want to keep them, or else to remove the logic that tries to ensure that ...

Again this is maybe something we could discuss a little more. I'll continue to familiarize myself more with the utility of having/not having the bound so that maybe I can offer an opinion instead of just asking what to do.